src/signature.rs

```diff
@@ -0,0 +1,71 @@
+//! Device electronic signature
+//!
+//! (stored in flash memory)
+
+use core::str::from_utf8;
+
+/// Uniqure Device ID register
+#[derive(Hash, Debug)]
+#[repr(C)]
+pub struct Uid {
+    x: u16,
+    y: u16,
+    waf_lot: [u8; 8],
+}
+
+impl Uid {
+    fn ptr() -> *const Self {
+        0x1FFF_7A10 as *const _
+    }
+
+    /// Returns a wrapped reference to the value in flash memory
+    pub fn new() -> &'static Self {
+        unsafe { &*Self::ptr() }
+    }
+
+    /// X coordinate on wafer
+    pub fn x(&self) -> u16 {
+        self.x
+    }
+
+    /// Y coordinate on wafer
+    pub fn y(&self) -> u16 {
+        self.y
+    }
+
+    /// Wafer number
+    pub fn waf_num(&self) -> u8 {
+        self.waf_lot[0].into()
+    }
+
+    /// Lot number
+    pub fn lot_num(&self) -> &str {
+        from_utf8(&self.waf_lot[1..]).unwrap()
```

Comment:
It might be better to use from_utf8_unchecked() rather than using unwrap() ? And it is supposed to be more lightweight as well, which might be important on a microcontroller.